src/accounts/context/userAccount.tsx

```diff
@@ -132,35 +128,39 @@ export const UserAccountProvider: FC<Props> = React.memo(({children}) => {
     }
   }
 
-  async function handleRenameActiveAccount(accountId, newName) {
-    const isActiveAcct = acct => acct.isActive
-    const activeIndex = userAccounts.findIndex(isActiveAcct)
-    const oldName = userAccounts[activeIndex].name
+  const handleRenameActiveAccount = useCallback(
+    async (accountId, newName) => {
+      const isActiveAcct = acct => acct.isActive
+      const activeIndex = userAccounts.findIndex(isActiveAcct)
+      const oldName = userAccounts[activeIndex].name
 
-    try {
-      const resp = await patchAccount({accountId, data: {name: newName}})
-
-      if (resp.status !== 200) {
-        dispatch(notify(accountRenameError(oldName)))
-      } else {
-        dispatch(notify(accountRenameSuccess(oldName, newName)))
+      try {
+        const accountData = await updateUserAccount(accountId, newName)
         event('multiAccount.renameAccount')
+        dispatch(notify(accountRenameSuccess(oldName, newName)))
 
         // change the name, and reset the active accts:
-        userAccounts[activeIndex].name = newName
-        setUserAccounts(userAccounts)
+        const updatedAccounts = userAccounts.map((account, idx) => {
+          const acct = {...account}
+          if (idx === activeIndex) {
+            acct.name = accountData.name
+          }
+
+          return acct
+        })
```

Comment:
deliberate usage of `map` here so that I am not modifying the state variable directly. refactored other parts.

Reply:
This is probably okay either way, as the old code set the `.name` property directly. But I'm also a little leery of directly manipulating context even if we're about to call setUserAccounts anyway. The O(n) from `map` isn't going to be expensive in a meaningful way, and following the usual pattern of not manipulating state directly keeps us out of trouble if we have to change this later. 

For example, we'll probably move the account state into redux eventually. And in redux we'll be foreclosed from making direct changes to state because the state will be frozen by immer.

Reply:
@hoorayimhelping Do you have a view on this one?

Reply:
The code here has changed since these comments were made, so I'm trying to get the context of the original discussion, but even without a ton of context, there are two things to consider here:

1. If we're thinking about the performance of this, keep in mind that using the spread operator is not a free operation. It likely has the same or slightly higher cost as a `O(n)` `map` operation, since it needs to iterate through all the elements of a collection and copy them (the copying operation accounting for the slightly higher cost)
2. All things (like performance and security) being equal, the best solution is the one that finds the balance between clarity and simplicity. If mapping over an array and changing a single item is necessary to call attention to, do it. If we can do the same thing without a loop, that is probably better

Reply:
updated!